Containerfile.base

```diff
@@ -0,0 +1,14 @@
+FROM registry.access.redhat.com/ubi9/go-toolset:1.21 as build
+WORKDIR /app
+COPY ./api api
+COPY ./cmd cmd
+COPY ./deploy deploy
+COPY ./hack hack
+COPY ./internal internal
+COPY ./go.* .
+COPY ./pkg pkg
+COPY ./test test
+COPY ./Makefile .
+
+USER 0
+RUN make build
```

Comment:
I know this is annoying, and we would replicate this code across all but, can we have all the container files self-contained (build stage + image creation?)

We could modify the "make build" to have single binary targets, which is something that bugged me for long, it makes no sense to build the client in there for example.

The advantage of single container files is that it's easier to audit or build by external people, to build in other downstream systems, quay, etc.

Reply:
OK